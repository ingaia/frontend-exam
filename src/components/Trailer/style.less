@import './../../variables.less';

.trailer {
    position: relative;
    height: 222px;
    width: 375px;
    margin: 30px;
    overflow: hidden;
    border: 1px solid @hillary;
    cursor: pointer;
    
    .thumbnail {
        height: 100%;
        width: 100%;
        background-repeat: no-repeat;
        background-size: cover;
        transform: scale(1.4);
    }
    
    .trailer-title {
        position: absolute;
        left: 0;
        right: 0;
        bottom: 0px;
        padding: 10px 5px;
        background-color: rgba(20,20,20, .7);
        color: #fff;
        text-align: center;
        pointer-events: none;
    }
}

@media @mobile {
    .trailer {
        margin: 10px;
        width: -webkit-fill-available;
    }
}

@media @tablet {
    .trailer {
        margin: 20px 10px;
    }
}

@media only screen and (max-width: 730px) and (min-width: 500px) {
    .trailer {
        height: 320px;
    }
}