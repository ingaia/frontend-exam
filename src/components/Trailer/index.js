import React, { useState } from 'react';

import VideoModal from '../VideoModal';

import './style.less';

export default function Trailer(props){
    const [showVideo, setShowVideo] = useState(false);

    return (
        <>
            <li className="trailer" onClick={() => setShowVideo(true)}>
                <div className="thumbnail" style={{ backgroundImage: `url(${props.thumbnail})` }}></div>
                <span className="trailer-title">{props.title}</span>
            </li>
            { showVideo ? <VideoModal videoId={props.videoId} closeVideo={() => setShowVideo(false)}/> : null}
        </>
    );
}