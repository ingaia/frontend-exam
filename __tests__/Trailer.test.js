import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json'
import Trailer from '../src/components/Trailer';

describe('<Trailer/>', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Trailer/>);
    });

    it('should render', () => {
        wrapper.setProps({
            label: 'https://i.ytimg.com/vi/1ke3zayKKm8/maxresdefault.jpg',
            title: 'The Witcher 3: Wild Hunt'
        });
        const trailer = wrapper.find('li');

        expect(trailer.hasClass('trailer')).toBe(true);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should have a title', () => {
        wrapper.setProps({
            title: 'The Witcher 3: Wild Hunt'
        });
        const trailer = wrapper.find('li');
        const title = trailer.find('.trailer-title');

        expect(title.text()).toBe('The Witcher 3: Wild Hunt');
    });
});