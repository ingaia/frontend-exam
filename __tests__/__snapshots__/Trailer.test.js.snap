// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Trailer/> should render 1`] = `
<Fragment>
  <li
    className="trailer"
    onClick={[Function]}
  >
    <div
      className="thumbnail"
      style={
        Object {
          "backgroundImage": "url(undefined)",
        }
      }
    />
    <span
      className="trailer-title"
    >
      The Witcher 3: Wild Hunt
    </span>
  </li>
</Fragment>
`;
